core/shared/src/main/scala/fs2/text.scala

```diff
@@ -542,9 +548,9 @@ object text {
             else Pull.output1(result)
           }
         case Some((chunk, stream)) =>
-          val linesBuffer = ArrayBuffer.empty[String]
+          linesBuffer.clear()
```

Comment:
Can we clear this earlier? To avoid holding on to lines after they've been emitted?

Reply:
I'll check, thx

Reply:
actually I've reverted this part back to creating a new array buffer for each output chunk 

it's a mutable array and it is not safe to clear it because of unexpected consequences if the chunk is used after we pull the next one

Reply:
Good catch! Thanks!